scylla/src/transport/session_test.rs

```diff
@@ -505,12 +505,17 @@ async fn test_token_calculation() {
                 .as_bigint()
                 .unwrap(),
         };
-        let expected_token = Murmur3Partitioner::hash(
+        let prepared_token = Murmur3Partitioner::hash(
             prepared_statement
                 .compute_partition_key(&serialized_values)
                 .unwrap(),
         );
-        assert_eq!(token, expected_token)
+        assert_eq!(token, prepared_token);
+        let cluster_data_token = session
+            .get_cluster_data()
+            .compute_token(&ks, "t3", &serialized_values)
+            .unwrap();
+        assert_eq!(token, cluster_data_token);
```

Comment:
Oh I didn't notice that there is this test as well.

My comment was originally about this test, on line `220`:
https://github.com/scylladb/scylla-rust-driver/blob/5a56e852763ee04e5883180b020810880aea0f93/scylla/src/transport/session_test.rs#L220

Please also extend the two test cases at `L220`. They test multiple partition key sizes, while this test only tests partition key of size 1, but with different string lengths.

Reply:
When I ran the tests, I've noticed that test keyspace metadata was not loaded, so `ClusterData::compute_token` cannot actually select the table partitioner and has to use the default one (Murmur3). I had to manually add `session.refresh_metadata().await.unwrap();` before calling `compute_token`. 

I believed that metadata refresh was automatic (I'd also tried `tokio::time::sleep(std::time::Duration::from_secs(10)).await;`, but there was no change, keyspace was still absent). Is there an issue ? Should I add `session.refresh_metadata().await.unwrap();` in the test before calling `compute_token`?

Reply:
The driver automatically awaits schema agreement, but it doesn't refresh cluster data immediately. It waits for all nodes to agree about the new schema version, but doesn't fetch it.
Topology data is refreshed every 60 seconds.

This was done to make sure that after creating a keyspace, table, etc we can use it without any errors, load balancing would catch up eventually.

It might be worth it to fetch the new schema immediately each time it changes, I've opened an issue: #544

Reply:
> Should I add session.refresh_metadata().await.unwrap(); in the test before calling compute_token?

Yes please add it.

Reply:
Done, sorry for the delay